zebrad/src/components/sync.rs

```diff
@@ -135,15 +209,25 @@ where
             AlwaysHedge,
             20,
             0.95,
-            2 * SYNC_RESTART_TIMEOUT,
+            2 * SYNC_RESTART_DELAY,
         );
+        // We apply a timeout to the verifier to avoid hangs due to missing earlier blocks.
+        let verifier = Timeout::new(verifier, BLOCK_VERIFY_TIMEOUT);
+        // Warn the user if we're ignoring their configured lookahead limit
+        let lookahead_limit = if config.sync.lookahead_limit < MIN_LOOKAHEAD_LIMIT {
+            tracing::warn!(config_lookahead_limit = ?config.sync.lookahead_limit, ?MIN_LOOKAHEAD_LIMIT,
+                          "configured lookahead limit too low: using minimum lookahead limit");
+            MIN_LOOKAHEAD_LIMIT
+        } else {
```

Comment:
I'm worried that users might not notice this warning, what do you think about turning this into a hard error at startup, probably by adding some sort of `fn validate(&self) -> Result<(), ConfigError>;` that we call in `ZebraApp` when we're initializing the config and what not.

Reply:
If all our components were abscissa FrameworkComponents, we could do config validation in the after_config callback, and return a FrameworkError:
https://docs.rs/abscissa_core/0.5.2/abscissa_core/application/trait.Application.html#tymethod.after_config

But we had trouble with component dependencies, so some components aren't abscissa FrameworkComponents.

Since this change isn't a high priority, and it isn't really the focus of this PR, I suggest that we change this PR to:
* panic, or
* revert the new warning to the previous code which silently ignored the config.

And then we open a follow-up task to create a consistent design for:
* defining limits for each config value
* enforcing those limits
* handing limit errors